### PR TITLE
Refactor frontmatter handling

### PR DESCRIPTION
**Pull request by duailibe**
We now have several methods for handling front-matter. This commit normalizes this code to be reused in different parsers.

When we introduce the pragma detection and insertion for CSS we'll need to use like Markdown does right now.

### CODE REVIEW

File: src/language-css/parser-postcss.js [hunk 1]
```diff
@@ -438,32 +439,32 @@ function parseNestedCSS(node) {
 }
 
 function parseWithParser(parser, text) {
+  const frontmatter = (function() {
+    const parsed = parseFrontmatter(text);
+    text = parsed.content;
+    return parsed.frontmatter;
+  })();
```
Comment (j-f1) [inline src/language-css/parser-postcss.js]: You should be able to remove the IIFE here.

File: src/language-css/parser-postcss.js [hunk 2]
```diff
@@ -438,32 +439,32 @@ function parseNestedCSS(node) {
 }
 
 function parseWithParser(parser, text) {
+  const frontmatter = (function() {
```
Comment (j-f1) [inline src/language-css/parser-postcss.js]: I personally would write this `frontMatter` and `front-matter`. What do you think?

File: src/language-css/parser-postcss.js [hunk 3]
```diff
@@ -472,7 +473,7 @@ function requireParser(isSCSS) {
   }
 
   // TODO: Remove this hack when this issue is fixed:
-  // https://github.com/shellscape/postcss-less/issues/88
+  // https://github.com/shellscape/postcss-less/isses/88
```
Comment (j-f1) [inline src/language-css/parser-postcss.js]: Typo?

File: src/utils/frontmatter.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    delimiter = "---";
+  }
+
+  if (text.indexOf("+++") === 0) {
+    delimiter = "+++";
+  }
+
+  let end = -1;
+
+  if (!delimiter || (end = text.indexOf(`\n${delimiter}`, 3)) === -1) {
```
Comment (j-f1) [inline src/utils/frontmatter.js]: You shouldn’t need the start index — if the file starts with `---`, it by definition doesn’t start with `\n---` and we’ll instead capture the whole document even without the start index.

File: tests/css_yaml/__snapshots__/jsfmt.spec.js.snap [hunk 5]
```diff
@@ -31,6 +31,9 @@ a {
 }
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ---
+
+
+
```
Comment (j-f1) [inline tests/css_yaml/__snapshots__/jsfmt.spec.js.snap]: This looks like a regression to me.

Comment (duailibe) [reply tests/css_yaml/__snapshots__/jsfmt.spec.js.snap]: We'll print the front(-)matter as-is from now on

File: src/language-css/parser-postcss.js (hunk 1 above)
Comment (duailibe) [reply src/language-css/parser-postcss.js]: I know that, but why?

File: src/language-css/parser-postcss.js (hunk 2 above)
Comment (duailibe) [reply src/language-css/parser-postcss.js]: Makes no difference to me 🤷‍♂️

File: src/utils/frontmatter.js (hunk 4 above)
Comment (duailibe) [reply src/utils/frontmatter.js]: > and we’ll instead capture the whole document even without the start index.

I don't understand what you mean. You're right it doesn't need the start index but that won't change anything

File: src/language-css/parser-postcss.js (hunk 1 above)
Comment (j-f1) [reply src/language-css/parser-postcss.js]: It feels like it would run slower, And it makes the code more complex and harder to understand.

**Pull request merged**